python/lsst/ts/m2gui/controltab/tab_alarm_warn.py

```diff
@@ -146,43 +216,90 @@ def _set_layout(self):
 
         self._button_reset = set_button("Reset All Items", self._callback_reset)
         self._button_enable_open_loop_max_limit = set_button(
-            "Enable OL Max Limits", self._callback_enable_open_loop_max_limit
+            "Enable Open-Loop Max Limits", self._callback_enable_open_loop_max_limit
         )
 
         layout = QVBoxLayout()
         layout.addWidget(self._table_error)
         layout.addWidget(self._text_error_cause)
+        layout.addWidget(QLabel("Limit Switch Status"))
+        layout.addWidget(self._get_widget_limit_switch())
         layout.addWidget(self._button_reset)
         layout.addWidget(self._button_enable_open_loop_max_limit)
 
         return layout
 
-    @Slot()
-    def _callback_reset(self):
-        """Callback of the reset button to reset all alarms and warnings."""
+    def _get_widget_limit_switch(self):
+        """Get the widget of limit switch.
 
-        self._text_error_cause.clear()
+        Returns
+        -------
+        scroll_area : `PySide2.QtWidgets.QScrollArea`
+            Widget of limit switch.
+        """
 
-        self.model.reset_errors()
+        widget = QWidget()
+
+        layout = QHBoxLayout(widget)
+        for ring in (Ring.B, Ring.C, Ring.D, Ring.A):
+            layout.addLayout(
```

Comment:
I won't convince you to do that now, but just to raise the possibility: PySide2 supports snake_case with from __feature__ import snake_case. See https://stackoverflow.com/questions/67206894/how-does-the-new-snake-case-feature-work-in-pyside6, https://www.pythonguis.com/faq/pyside2-vs-pyside6/. Shall work in pyside2. As we try to comply to PEP8, that shall solve a lot of problems.. I will probably do that for ts_cRIOpy sometime soon.